swap/swap.go

```diff
@@ -147,68 +141,37 @@ func (s *Swap) DeploySuccess() string {
 func (s *Swap) Add(amount int64, peer *protocols.Peer) (err error) {
 	s.accountingLock.Lock()
```

Comment:
So in my understanding, we don't need `accountingLock` at all in this design.

We would move everything to `swapPeer`. Something like this:
```
func (s *Swap) Add(amount int64, peer *protocols.Peer) (err error) {
   swapPeer, ok := s.getPeer(peer.ID())   // here getPeer is (read-)locked by peersLock
   if !ok {
         ....
   }
   return swapPeer.add(amount)
```
That's it, then we implement `add` in `Peer` and move everything (including the threshold check and sending cheques) there, as it seems to me all other actions are done on the peer itself. This `add` function would be locked by the peer's `lock`.

Am I oversimplifying? Not seeing something? Please question and think hard.

Reply:
Not sure yet. We do access some fields on the shared `Swap` object but we only read them and they never change after the constructor.